### PR TITLE
KCL-6892 Data-attributes for plus button

### PR DESCRIPTION
**Pull request by vladislavb-kentico**
- Implemented several new data-attributes for add button.
- Implemented plus button rendering and positioning logic.
- Moved positioning logic from renderer to components to make it more readable.
- Fixed tests and storybook.

### CODE REVIEW

File: src/utils/customElements.ts [hunk 1]
```diff
@@ -1,3 +1,89 @@
...
+  }
+  // else treat node as content item if it has item id attribute
+  else if (element.hasAttribute(DataAttribute.ItemId)) {
+    return HighlightType.ContentItem;
+  }
+
+  return HighlightType.None;
+}
+
 export function createTemplateForCustomElement(html: string): HTMLTemplateElement {
```
Comment (PeterSkoda) [inline src/utils/customElements.ts:87]: just thinking if this function should be with the rest functions in the same file. It is just something different. But no biggie.

File: src/web-components/KSLHighlightElement.ts [hunk 2]
```diff
@@ -166,36 +156,68 @@ export class KSLHighlightElement extends KSLCustomElement {
...
   public disconnectedCallback(): void {
+    super.connectedCallback();
+
     this.editButtonRef.addEventListener('click', this.handleEditButtonClick);
     this.removeButtonRef.addEventListener('click', this.handleRemoveButtonClick);
     this.unregisterTargetNodeListeners();
   }
 
+  public attributeChangedCallback(attributeName: string, _oldValue: string | null, newValue: string | null): void {
+    if (attributeName === 'deleted') {
```
Comment (PeterSkoda) [inline src/web-components/KSLHighlightElement.ts:174]: wouldn't it be better to have these attributes as some constants? I see them there and there

File: src/web-components/abstract/KSLCustomElementWithTooltip.ts [hunk 3]
```diff
@@ -1,44 +1,34 @@
 import { KSLCustomElement } from './KSLCustomElement';
-import { ElementPosition } from './KSLPositionedElement';
 import { KSLTooltipElement } from '../KSLTooltipElement';
+import { ElementPositionOffset } from './KSLPositionedElement';
 
 export abstract class KSLCustomElementWithTooltip extends KSLCustomElement {
+  public static get observedAttributes(): string[] {
+    return ['tooltip-disabled', 'tooltip-message'];
```
Comment (PeterSkoda) [inline src/web-components/abstract/KSLCustomElementWithTooltip.ts:7]: just curious, have you thought about constants?

**Review by PeterSkoda — APPROVED** [review]
Approved with some comments

File: src/web-components/abstract/KSLCustomElementWithTooltip.ts (hunk 3 above)
Comment (vladislavb-kentico) [reply src/web-components/abstract/KSLCustomElementWithTooltip.ts:7]: We discussed this on the call. We will leave attribute names as strings for now and possibly replace them with constants in the future.

**Pull request merged**